app/backend/src/couchers/servicers/requests.py

```diff
@@ -635,6 +642,28 @@ def MarkLastSeenHostRequest(self, request, context, session):
         session.commit()
         return empty_pb2.Empty()
 
+    def ArchiveHostRequest(self, request, context, session):
+        host_request = session.execute(
+            select(HostRequest).where(HostRequest.conversation_id == request.host_request_id)
+        ).scalar_one_or_none()
```

Comment:
Please use the same query as in `GetHostRequest`; maybe we should move it into a helper function given it's a bit hectic.